nixos/modules/services/networking/seafile.nix

```diff
@@ -0,0 +1,272 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  python = pkgs.python3Packages.python;
+  cfg = config.services.seafile;
+
+  ccnetConf = pkgs.writeText "ccnet.conf" ''
+    [General]
+    SERVICE_URL = ${cfg.serviceUrl}
+  '';
+
+  seafileConf = pkgs.writeText "seafile.conf" ''
+    [fileserver]
+    port=${toString cfg.fileserverPort}
+    host=${cfg.fileserverHost}
+    ${cfg.seafileExtraConf}
+  '';
+
+  seahubSettings = pkgs.writeText "seahub_settings.py" ''
+    FILE_SERVER_ROOT = '${cfg.serviceUrl}/seafhttp'
+    DATABASES = {
+        'default': {
+            'ENGINE': 'django.db.backends.sqlite3',
```

Comment:
I'm not sure what this is used for, but is there any other database option, and if so, would it make sense to allow it to be configurable?

Reply:
Seafile also supports mysql, but I don't plan to use it right now. There may be postgresql support at some point, which I'd be more interested in.
I thought extra DB support could come as another PR?

Reply:
Yeah. That sounds good :+1: